packages/teams-js/test/public/media.spec.ts

```diff
@@ -838,7 +713,146 @@ describe('media', () => {
           },
         } as DOMMessageEvent);
 
-        return expect(promise).rejects.toEqual({ errorCode: ErrorCode.SIZE_EXCEEDED });
+        await expect(promise).rejects.toMatchObject({ errorCode: ErrorCode.SIZE_EXCEEDED });
+      });
+    });
+  });
+
+  describe('videoController', () => {
+    describe('v1', () => {
+      it('videoController notifyEventToHost is handled successfully', async () => {
+        expect.assertions(7); // initializeWithContext has 3 assertions + 4 in this test = 7
+
+        await mobilePlatformMock.initializeWithContext(FrameContexts.task, HostClientType.android);
+        mobilePlatformMock.setClientSupportedSDKVersion(nonFullScreenVideoModeAPISupportVersion);
+        const sendMessageToParentSpy = jest.spyOn(communication, 'sendMessageToParent');
+
+        await expect(new media.VideoController().stop(emptyCallback)).resolves.not.toThrow();
+
+        const message = mobilePlatformMock.findMessageByFunc('media.controller');
+        expect(message).not.toBeNull();
+        expect(message.args.length).toBe(1);
+        const callbackId = message.id;
+
+        mobilePlatformMock.respondToMessage({
+          data: {
+            id: callbackId,
+            args: [undefined],
+          },
+        } as DOMMessageEvent);
+
+        expect(sendMessageToParentSpy).toHaveBeenCalled();
+      });
+
+      it('videoController notifyEventToHost is not handled successfully', async () => {
+        expect.assertions(7);
+
+        await mobilePlatformMock.initializeWithContext(FrameContexts.content, HostClientType.android);
+        mobilePlatformMock.setClientSupportedSDKVersion(nonFullScreenVideoModeAPISupportVersion);
+
+        const sendMessageToParentSpy = jest.spyOn(communication, 'sendMessageToParent');
+        const err = {
+          errorCode: ErrorCode.INTERNAL_ERROR,
+        };
+
+        await new media.VideoController().stop((e: SdkError) => {
+          expect(e).toEqual(err);
+        });
+
+        const message = mobilePlatformMock.findMessageByFunc('media.controller');
+        expect(message).not.toBeNull();
+        expect(message.args.length).toBe(1);
+        const callbackId = message.id;
+
+        mobilePlatformMock.respondToMessage({
+          data: {
+            id: callbackId,
+            args: [err],
+          },
+        } as DOMMessageEvent);
+
+        expect(sendMessageToParentSpy).toHaveBeenCalled();
+      });
+
+      it('videoController notifyEventToHost should fail in default version of platform and should exit early', async () => {
+        expect.assertions(5);
+
+        await mobilePlatformMock.initializeWithContext(FrameContexts.content, HostClientType.android);
+        mobilePlatformMock.setClientSupportedSDKVersion(originalDefaultPlatformVersion);
+
+        const sendMessageToParentSpy = jest.spyOn(communication, 'sendMessageToParent');
+
+        try {
+          await new media.VideoController().stop(e => {
```

Comment:
I'm curious how this works. By looking at your code, it seems like if we pass the callback in then we always return the error through the callback (which is correct), but in this test it seems to be expecting and catching the exception on 790. Given that you are expecting 5 assertions on 778 (3 in initializeWithContext and 2 in this test) clearly it's getting hit I'm just not sure I understand why

Reply:
![image](https://user-images.githubusercontent.com/9360417/152590482-fe8a2943-93c7-45f2-8c47-76f89cce0bfc.png)
 this is where i got the information from. If i checked in the callback, it wouldn't wait for the callback to execute and the assertions it would expect would be 4.  When i would write that I expected it to reject, it would say it threw an error, but the test wouldn't pass, i'm actually not sure why this happened.

Reply:
when i was referring to testing the thrown error  that i am confused by, what i wrote was 
```
await expect(new media.VideoController().stop(e=>e))).rejects.toEqual({old platform error})
```